In [67]:
import csv
import agate

In [68]:
tester = agate.TypeTester()

In [69]:
standards = agate.Table.from_csv('english.csv', tester)

In [74]:
six_years_not_met = standards.where(lambda row: row['CONSECUTIVE_YEARS_NOT_MET'] == 6)
six_years_not_met.print_table()

|-----------------+-----------------+--------------------------+-----------+-----------------+-----------+-----------+-----------+----------+---------------------------+----------------------------------------|
|  YEAR_TITLE_III | COUNTY_DISTRICT | COUNTY_DISTRICT_NAME     | TITLE_III | YEAR_ASSESSMENT | AMAO1_MET | AMAO2_MET | AMAO3_MET | AMAO_MET | CONSECUTIVE_YEARS_NOT_MET | TITLE_III_STATUS                       |
|-----------------+-----------------+--------------------------+-----------+-----------------+-----------+-----------+-----------+----------+---------------------------+----------------------------------------|
|           2,014 |           5,120 | WHEATON R-III            |      True |           2,013 | MET       | MET       | NOT MET   | NOT MET  |                         6 | Title III District Improvement Year 4  |
|           2,014 |          10,093 | COLUMBIA 93              |      True |           2,013 | MET       | MET       | NOT MET   | NOT MET  |               

In [82]:
no_sanctions = standards.where(lambda row: row['TITLE_III_STATUS'] == 'Title III No Sanctions').where(lambda row: row['YEAR_ASSESSMENT'] == 2014)
no_sanctions.print_table()

|-----------------+-----------------+----------------------+-----------+-----------------+-----------+-----------+-----------+----------+---------------------------+-------------------------|
|  YEAR_TITLE_III | COUNTY_DISTRICT | COUNTY_DISTRICT_NAME | TITLE_III | YEAR_ASSESSMENT | AMAO1_MET | AMAO2_MET | AMAO3_MET | AMAO_MET | CONSECUTIVE_YEARS_NOT_MET | TITLE_III_STATUS        |
|-----------------+-----------------+----------------------+-----------+-----------------+-----------+-----------+-----------+----------+---------------------------+-------------------------|
|           2,015 |          96,102 | CLAYTON              |      True |           2,014 | MET       | MET       | MET       | MET      |                         0 | Title III No Sanctions  |
|-----------------+-----------------+----------------------+-----------+-----------------+-----------+-----------+-----------+----------+---------------------------+-------------------------|


In [73]:
num_schools_six_years_not_met = len(six_years_not_met.rows)
print num_schools_six_years_not_met

45


In [76]:
schools_not_meeting_AMAO1_2014 = standards.where(lambda row: row['AMAO1_MET'] == "NOT MET" ).where (lambda row: row ['YEAR_ASSESSMENT'] == 2014)
schools_not_meeting_AMAO1_2014.print_table()

|-----------------+-----------------+-------------------------+-----------+-----------------+-----------+-----------+-----------+----------+---------------------------+-------------------|
|  YEAR_TITLE_III | COUNTY_DISTRICT | COUNTY_DISTRICT_NAME    | TITLE_III | YEAR_ASSESSMENT | AMAO1_MET | AMAO2_MET | AMAO3_MET | AMAO_MET | CONSECUTIVE_YEARS_NOT_MET | TITLE_III_STATUS  |
|-----------------+-----------------+-------------------------+-----------+-----------------+-----------+-----------+-----------+----------+---------------------------+-------------------|
|           2,015 |           2,097 | SAVANNAH R-III          |           |           2,014 | NOT MET   | NOT MET   |           | NOT MET  |                         0 |                   |
|           2,015 |          10,089 | HALLSVILLE R-IV         |           |           2,014 | NOT MET   | NOT MET   |           | NOT MET  |                         0 |                   |
|           2,015 |          14,129 | FULTON 58        

In [77]:
num_schools_not_meeting_AMAO1_2014 = len(schools_not_meeting_AMAO1_2014.rows)
print num_schools_not_meeting_AMAO1_2014

53


In [78]:
schools_not_meeting_AMAO2_2014 = standards.where(lambda row: row['AMAO2_MET'] == "NOT MET" ).where (lambda row: row ['YEAR_ASSESSMENT'] == 2014)
schools_not_meeting_AMAO2_2014.print_table()

|-----------------+-----------------+--------------------------------+-----------+-----------------+-----------+-----------+-----------+----------+---------------------------+----------------------------------------|
|  YEAR_TITLE_III | COUNTY_DISTRICT | COUNTY_DISTRICT_NAME           | TITLE_III | YEAR_ASSESSMENT | AMAO1_MET | AMAO2_MET | AMAO3_MET | AMAO_MET | CONSECUTIVE_YEARS_NOT_MET | TITLE_III_STATUS                       |
|-----------------+-----------------+--------------------------------+-----------+-----------------+-----------+-----------+-----------+----------+---------------------------+----------------------------------------|
|           2,015 |           2,090 | AVENUE CITY R-IX               |           |           2,014 | MET       | NOT MET   |           | NOT MET  |                         0 |                                        |
|           2,015 |           2,097 | SAVANNAH R-III                 |           |           2,014 | NOT MET   | NOT MET   |        

In [79]:
num_schools_not_meeting_AMAO2_2014 = len(schools_not_meeting_AMAO2_2014.rows)
print num_schools_not_meeting_AMAO2_2014

175


In [80]:
schools_not_meeting_AMAO3_2014 = standards.where(lambda row: row['AMAO3_MET'] == "NOT MET" ).where (lambda row: row ['YEAR_ASSESSMENT'] == 2014)
schools_not_meeting_AMAO3_2014.print_table()

|-----------------+-----------------+--------------------------------+-----------+-----------------+-----------+-----------+-----------+----------+---------------------------+----------------------------------------|
|  YEAR_TITLE_III | COUNTY_DISTRICT | COUNTY_DISTRICT_NAME           | TITLE_III | YEAR_ASSESSMENT | AMAO1_MET | AMAO2_MET | AMAO3_MET | AMAO_MET | CONSECUTIVE_YEARS_NOT_MET | TITLE_III_STATUS                       |
|-----------------+-----------------+--------------------------------+-----------+-----------------+-----------+-----------+-----------+----------+---------------------------+----------------------------------------|
|           2,015 |           4,110 | MEXICO 59                      |      True |           2,014 | MET       | NOT MET   | NOT MET   | NOT MET  |                         6 | Title III District Improvement Year 4  |
|           2,015 |           5,120 | WHEATON R-III                  |           |           2,014 | MET       | NOT MET   | NOT MET

In [81]:
num_schools_not_meeting_AMAO3_2014 = len(schools_not_meeting_AMAO3_2014.rows)
print num_schools_not_meeting_AMAO3_2014

83


In [83]:
schools_meeting_AMAO3_2014 = standards.where(lambda row: row['AMAO3_MET'] == "MET" ).where (lambda row: row ['YEAR_ASSESSMENT'] == 2014)
schools_meeting_AMAO3_2014.print_table()

|-----------------+-----------------+----------------------+-----------+-----------------+-----------+-----------+-----------+----------+---------------------------+-------------------------|
|  YEAR_TITLE_III | COUNTY_DISTRICT | COUNTY_DISTRICT_NAME | TITLE_III | YEAR_ASSESSMENT | AMAO1_MET | AMAO2_MET | AMAO3_MET | AMAO_MET | CONSECUTIVE_YEARS_NOT_MET | TITLE_III_STATUS        |
|-----------------+-----------------+----------------------+-----------+-----------------+-----------+-----------+-----------+----------+---------------------------+-------------------------|
|           2,015 |          96,102 | CLAYTON              |      True |           2,014 | MET       | MET       | MET       | MET      |                         0 | Title III No Sanctions  |
|-----------------+-----------------+----------------------+-----------+-----------------+-----------+-----------+-----------+----------+---------------------------+-------------------------|


Columbia Public Schools haven't met Missouri’s Annual Measurable Achievement Objectives for English Language Proficiency for six consecutive years. 

Annual Measurable Achievement Objectives (AMAOs) are standards for English Language Learner students set by the state as required by federal mandate. There are three specific goals districts must meet to pass the standards.

•	AMAO 1: calculated by the number of students showing improvement in learning English.
•	AMAO 2: calculated by the number of students achieving proficiency in English.
•	AMAO 3: calculated by English Language Learner scores on communication arts and mathematics standardized tests.(The standardized test in Missouri is the MAP. Students who have been in the country for more than a year are required to take the full MAP test regardless of their level of English language proficiency.)

For every year that school districts fail to meet the AMAOs, state sanctions increase in severity. If a school fails any one of the three goals, they are considered to fail the standards. 

If a district doesn't reach the standards for four years or more, the state can technically remove federal funding (TITLE III) for English Language Learner students, require the district to change its curriculum, or ask the district to remove any personnel responsible for the failure.

So far, the state has not taken any action against any school districts not meeting the standards – even those districts that continuously fail to meet the standards for multiple years. 

In assessment year 2014 (the most recently available school year), only one school district, Clayton, met all three AMAO goals listed above, thereby passing the standards. 

Columbia was one of 45 school districts to fail the standards for the 6th year consecutively. 
Apparently, some standards are harder to meet than others – in 2014, 53 school districts failed AMAO1, 175 school districts failed AMAO2, and 83 school districts failed AMAO3. 

It’s important to remember that for AMAO1, 2 or 3, some school districts have too small a population of English language learners (or students who are too new to the system to be tested). In the data, those districts are noted as NA or blank in the columns for AMAO1, 2 or 3. 

It appears that at least in 2014, AMAO3 was the most difficult goal to reach – only Clayton met AMAO3 and successfully avoided any Title III Sanctions (or the threat of them at least!).
